src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java

```diff
@@ -242,6 +242,12 @@ public int buildingRequiresCertainAmountOfItem(final ItemStack stack, final List
         {
             return stack.getCount();
         }
+
+        if (jobEntry == ModJobs.cookassistant)
+        {
+            return stack.getCount();
+        }
+
```

Comment:
Don't hardcode this.
Make an interface for this and check if the job has that interface implemented.

Reply:
I think the proper way when this stuff is job dependent, not building, is to forward the call to each assigned citizen instead

Reply:
Or to the affected citizen at least, as this method is called from a context where a citizen is known (that is where jobEntry is coming from)

Reply:
I don't care particularly how this is solved, as long as it is not hardcoded.

Reply:
if it is only going from citizen to building to citizen, then it may aswell just stay within the citizen instead, indeed

Reply:
I can't really forward this call to the citizen or job. This is the only instance. And the cook building decides how the two cooks behave. This is okay, this is not in a generic high level class.

Reply:
Ofc you could forward the call to the citizens job. 
`This is the only instance` what do you mean with that what exactly is the "only instance"?
Why is it the cook building required to decide this behaviour when apparently you need job-selective behaivour?

Reply:
I'm not gonna move a shitton of code around into the jobs to fix this little bug that only affects the cook assistant.

Reply:
I understand not wanting to have this in AbstractBuilding, but this is in the BuildingCook

Reply:
ah thats why your answer didnt make a lot of sense, it was "mimimi" dont wanna change it xD

Reply:
okay change it yourself.